flink-connector-pinot/src/main/java/org/apache/flink/streaming/connectors/pinot/PinotSink.java

```diff
@@ -0,0 +1,361 @@
+        this.fsAdapter = checkNotNull(fsAdapter);
+        checkArgument(numCommitThreads > 0);
+        this.numCommitThreads = numCommitThreads;
+    }
+
+    /**
+     * Creates a Pinot sink writer.
+     *
+     * @param context InitContext
+     * @param states  Empty list as the PinotSinkWriter does not accept states.
+     */
+    @Override
+    public PinotSinkWriter<IN> createWriter(InitContext context, List<Void> states) {
+        return new PinotSinkWriter<>(
+                context.getSubtaskId(), maxRowsPerSegment, eventTimeExtractor,
+                jsonSerializer, fsAdapter
+        );
+    }
+
+    /**
+     * The PinotSink does not use a committer. Instead a global committer is used
+     *
+     * @return Empty Optional
+     */
+    @Override
+    public Optional<Committer<PinotSinkCommittable>> createCommitter() {
+        return Optional.empty();
+    }
+
+    /**
+     * Creates the global committer.
+     */
+    @Override
+    public Optional<GlobalCommitter<PinotSinkCommittable, PinotSinkGlobalCommittable>> createGlobalCommitter() throws IOException {
+        String timeColumnName = eventTimeExtractor.getTimeColumn();
+        TimeUnit segmentTimeUnit = eventTimeExtractor.getSegmentTimeUnit();
+        PinotSinkGlobalCommitter committer = new PinotSinkGlobalCommitter(
+                pinotControllerHost, pinotControllerPort, tableName, segmentNameGenerator,
+                tempDirPrefix, fsAdapter, timeColumnName, segmentTimeUnit, numCommitThreads
+        );
+        return Optional.of(committer);
+    }
+
+    /**
+     * Creates the committables' serializer.
+     */
+    @Override
+    public Optional<SimpleVersionedSerializer<PinotSinkCommittable>> getCommittableSerializer() {
+        return Optional.of(new PinotSinkCommittableSerializer());
+    }
+
+    /**
+     * Creates the global committables' serializer.
+     */
+    @Override
+    public Optional<SimpleVersionedSerializer<PinotSinkGlobalCommittable>> getGlobalCommittableSerializer() {
+        return Optional.of(new PinotSinkGlobalCommittableSerializer());
+    }
+
+    /**
+     * The PinotSink does not use writer states.
+     *
+     * @return Empty Optional
+     */
+    @Override
+    public Optional<SimpleVersionedSerializer<Void>> getWriterStateSerializer() {
+        return Optional.empty();
+    }
+
+    /**
+     * Builder for configuring a {@link PinotSink}. This is the recommended public API.
+     *
+     * @param <IN> Type of incoming elements
+     */
+    public static class Builder<IN> {
+        String pinotControllerHost;
+        String pinotControllerPort;
+        String tableName;
+        int maxRowsPerSegment;
+        String tempDirPrefix = "flink-connector-pinot";
+        JsonSerializer<IN> jsonSerializer;
+        EventTimeExtractor<IN> eventTimeExtractor;
+        SegmentNameGenerator segmentNameGenerator;
+        FileSystemAdapter fsAdapter;
+        int numCommitThreads = 4;
```

Comment:
Nit: Use a constant variable `static final ...` for the magic number `4` and call it something like `DEFAULT_COMMIT_THREADS`